email-reader/src/test/java/no/sikt/nva/email/reader/handler/ScopusEmailReaderTest.java

```diff
@@ -0,0 +1,182 @@
+    }
+
+    @Test
+    void shouldThrowExceptionIfEmailDoesNotHaveContentTypeMultipart() throws IOException, MimeException {
+        var s3EventNotFromSikt = createS3Event(EmailGenerator.generateNonMultipartEmail());
+        var exception = assertThrows(EmailException.class, () -> handler.handleRequest(s3EventNotFromSikt, CONTEXT));
+        assertThat(exception.getMessage(), containsString(COULD_NOT_PARSE_EMAIL));
+        assertThat(exception.getObjectKey(), is(equalTo(extractObjectKey(s3EventNotFromSikt))));
+        assertThat(exception.getBucket(), is(equalTo(INPUT_BUCKET_NAME)));
+    }
+
+
+    @Test
+    void shouldThrowExceptionIfEmailDoesNotContainBodyWithURI() throws IOException, MimeException {
+        var s3EventNotFromSikt = createS3Event(EmailGenerator.generateEmailWithoutScopusLinks());
+        var exception = assertThrows(EmailException.class, () -> handler.handleRequest(s3EventNotFromSikt, CONTEXT));
+        assertThat(exception.getMessage(), containsString(NO_URL_PRESENT_IN_MESSAGE));
+        assertThat(exception.getObjectKey(), is(equalTo(extractObjectKey(s3EventNotFromSikt))));
+        assertThat(exception.getBucket(), is(equalTo(INPUT_BUCKET_NAME)));
+    }
+
+    @Test
+    void shouldNotDownloadCitedByUri() throws IOException {
+        var s3EventNotFromSikt = createS3Event(validEmail);
+        var actualUrl = handler.handleRequest(s3EventNotFromSikt, CONTEXT);
+        assertThat(actualUrl, not(contains(UriWrapper.fromUri("https://sccontent-scudd-delivery-prod.s3.amazonaws.com/sccontent-scudd-delivery-prod/cristin_1/2023-6-14/cristin_1_2023-6-14_ANI-CITEDBY.zip?my-query-param=should-be-preserved").getUri())));
+    }
+
+    //TODO: implement download feature.
+//    @Test
+//    void shouldThrowExceptionIfTheURIsInTheEmailIsNotDownloadable() throws IOException {
+//        var s3EventNotFromSikt = createS3Event(validEmail);
+//        var exception = assertThrows(EmailException.class, () -> handler.handleRequest(s3EventNotFromSikt, CONTEXT));
+//        assertThat(exception.getMessage(), containsString(NO_URL_PRESENT_IN_MESSAGE));
+//        assertThat(exception.getObjectKey(), is(equalTo(extractObjectKey(s3EventNotFromSikt))));
+//        assertThat(exception.getBucket(), is(equalTo(INPUT_BUCKET_NAME)));
+//
+//    }
+
+    @Test
+    void shouldReturnASetOfDownloadsURIWhenTheEmailContainsTheURI() throws IOException {
+        var s3EventNotFromSikt = createS3Event(validEmail);
+        var actualUrl = handler.handleRequest(s3EventNotFromSikt, CONTEXT);
+        var expectedUrls = urlsInValidEmailTxt();
+        assertThat(actualUrl, containsInAnyOrder(expectedUrls.toArray()));
+    }
+
+    private Set<URI> urlsInValidEmailTxt() {
+        return Set.of(UriWrapper.fromUri("https://sccontent-scudd-delivery-prod.s3.amazonaws.com/sccontent-scudd-delivery-prod/cristin_1/2023-6-14/cristin_1_2023-6-14_ANI-ITEM-full-format-xml.zip?my-query-param=should-be-preserved").getUri(),
```

Comment:
Not sure if it is a good idea to have the s3 uri directly in code, but i need another opinion. Maybe it is needed for testing

Reply:
washed it a bit more. (it was washed somewhat).
Want to keep part of the url, so it will be harder to trigger the lambda to download rubbish.